### ISSUE TITLE
Human position awareness

### ISSUE THREAD

**Issue by Alex18947** [duanzhiihao/RAPiD#48]
Hi, thanks for this very interesting project. I am studying the rotated bounding box concept and learned about the 90<=angle<90 restriction as well as the w<h restriction. This, of course, leads to the situation, that people with "opposite"orientation have the same bounding box angle. E.g. the two marked persons in this image will both roughly have the angle of 30 degrees (image taken from Cepfdof dataset):

![Image](https://github.com/user-attachments/assets/ba30e863-47ac-4e67-9301-87557ac897c9)

I am trying to understand whether the model (probably after some modifications) could actually benefit from knowing the "natural" rotation angle of the people during training, i.e. 30 and e.g. 210 degrees in this case leading to a potential better overall mAP.

**Comment by duanzhiihao** [comment]
Hi, thanks for your interest. The point you make is very interesting. As you noticed, we did not label the "natural rotation angle of the people" in the CEPDOF dataset. I think labeling it would probably not give a better mAP, because mAP only cares about the IoU between bounding boxes but not the "natural orientation" of the objects.

That said, I agree that knowing the natural rotation angle could be useful for some applications. I guess this could be achieved by predicting people's head locations in addition to the bounding boxes.

**Comment by Alex18947** [comment]
Thanks. Understood. One more thing I was wonderring about after reviewing the Coco pretraining procedure, what's the idea behind  having just one single image in the validation set ([here](https://github.com/duanzhiihao/RAPiD/tree/master/images/tiny_val)) ?

Also, have you tried to train the model on fisheye images from scratch i.e. with no Coco checkpoint? When combining all the fish-eye datasets the model could maybe have enough data for training and validating. Of course, this would require some code modifications (like [here](https://github.com/duanzhiihao/RAPiD/blob/master/train.py#L212), but I am thinking of giving it a try...

**Comment by duanzhiihao** [comment]
> Thanks. Understood. One more thing I was wonderring about after reviewing the Coco pretraining procedure, what's the idea behind having just one single image in the validation set ([here](https://github.com/duanzhiihao/RAPiD/tree/master/images/tiny_val)) ?

Hi, this single-image validation set is for debugging/unit testing only. We did not use it for validating model performance. Sry for the confusion.

> Also, have you tried to train the model on fisheye images from scratch i.e. with no Coco checkpoint? When combining all the fish-eye datasets the model could maybe have enough data for training and validating. Of course, this would require some code modifications (like [here](https://github.com/duanzhiihao/RAPiD/blob/master/train.py#L212), but I am thinking of giving it a try...

We indeed tried training models on fisheye images from scratch. That gives worse performance than using a COCO pre-trained model (I cannot recall the numbers exactly, but it was like around 10-15% AP@50 IoU drop). So we concluded that pre-training was critical for data-limited tasks like fisheye images.

**Comment by Alex18947** [comment]
Thank you. We will try to use a larger combined fish-eye dataset and compare the results when using a coco pretrained model vs. a model trained from scratch. In your paper, you don't seem to mention the coco pretraining procedure in detail. For example, during coco pretraining, what validation image set were you using? The one from coco or some fish-eye images?
